engines/osxnative/sandbox.go

```diff
@@ -0,0 +1,68 @@
+package macengine
+
+import (
+	"github.com/taskcluster/taskcluster-worker/engines"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"io"
+	"os"
+	"os/exec"
+)
+
+type sandbox struct {
+	context     *runtime.TaskContext
+	taskPayload *payload
+	env         *[]string
+	aborted     bool
+}
+
+func newSandbox(context *runtime.TaskContext, taskPayload *payload, env *[]string) *sandbox {
+	return &sandbox{
+		context,
+		taskPayload,
+		env,
+		false,
+	}
+}
+
+func (s *sandbox) WaitForResult() (engines.ResultSet, error) {
+	if s.aborted {
+		return resultset{success: false}, engines.ErrSandboxAborted
+	}
+
+	// Use the host environment plus custom environment variables
+	env := os.Environ()
+	env = append(env, *s.env...)
+
+	for _, command := range s.taskPayload.Command {
+		cmd := exec.Command(command[0], command[1:]...)
+		cmd.Env = env
+
+		output, err := cmd.CombinedOutput()
```

Comment:
So I haven't dealt with cmd much, but will this stream to the task log? We want the command streaming so live logging can be enabled rather than writing at the end.